snippets/base/models.py

```diff
@@ -258,7 +262,35 @@ def __unicode__(self):
         return self.description
 
 
-class Snippet(CachingMixin, models.Model):
+class SnippetBaseModel(models.Model):
+    def duplicate(self):
+        snippet_copy = copy.copy(self)
+        snippet_copy.id = None
+        snippet_copy.disabled = True
+        snippet_copy.name = '{0} - {1}'.format(
+            self.name,
+            datetime.strftime(datetime.now(), '%Y.%m.%d %H:%M:%S'))
+        snippet_copy.save()
+
+        for field in self._meta.get_all_field_names():
```

Comment:
I see what you mean and in a way a follow it by providing a semi-generic approach: `id`, `disabled`, `name` we very snippet specific and I didn't bother making this generic. 

But I opted for generic copy approach when it comes to the related models because I don't want to see in the future this action to make incomplete copies because we added `FooMatchRules` and we forgot to update it.